akka-cluster/src/main/scala/akka/cluster/VectorClock.scala

```diff
@@ -26,12 +26,19 @@ private[cluster] object VectorClock {
 
     def fromHash(hash: String): Node = hash
 
-    private def hash(name: String): String = {
+    private[this] def hash(name: String): String = {
       val digester = MessageDigest.getInstance("MD5")
       digester.update(name.getBytes("UTF-8"))
-      digester.digest.map { h =>
-        "%02x".format(0xFF & h)
-      }.mkString
+      val digest = digester.digest
+      val ch = new Array[Char](digest.length * 2)
+      var i = 0
+      while (i < digest.length) {
+        val h = digest(i)
+        ch(i * 2) = Character.forDigit((h >> 4) & 0xF, 16)
+        ch(i * 2 + 1) = Character.forDigit(h & 0xF, 16)
+        i += 1
+      }
+      String.valueOf(ch)
     }
```

Comment:
I don't think this is frequently used. I could only find that it's used when merging two conflicting Gossip messages and something about member removals, which shouldn't be very frequent.

Additionally, when testing this in the full context of merging two conflicting `VersionVector` I can't see any significant difference when changing this.

Therefore I'm thinking that we should not merge this even though it hurts me to reject a nice contribution. What do you think?

Reply:
My thoughts as well, but I wanted to not be the first to add it. This is a wonderful contribution but given the other factors I think it is better to not :( Though it highly appreciated.

Reply:
@412b Thanks for your effort and time on this, greatly appreciated indeed and well written. We'll have to go with the team thoughts noted above. Cheers and we look forward to more of you contribs !